src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -763,6 +770,38 @@ public boolean accept(File dir, String name) {
             }
         }
 
+        @edu.umd.cs.findbugs.annotations.SuppressWarnings(
+                value = {"RV_RETURN_VALUE_IGNORED_BAD_PRACTICE", "IS2_INCONSISTENT_SYNC"},
+                justification = "RV_RETURN_VALUE_IGNORED_BAD_PRACTICE=Best effort, " +
+                        "IS2_INCONSISTENT_SYNC=only called from an already synchronized method"
+        )
+        private void cleanupOldBundles(File bundleDir, File justGenerated) {
+            thread.setName(String.format("%s periodic bundle generator: tidying old bundles since %s",
+                    SupportPlugin.class.getSimpleName(), new Date()));
+            File[] files = bundleDir.listFiles(new FilenameFilter() {
+                public boolean accept(File dir, String name) {
+                    return name.endsWith(".zip");
+                }
+            });
+            long pivot = System.currentTimeMillis();
+            for (long l = 1; l * 2 > 0; l *= 2) {
+                boolean seen = false;
+                for (File f : files) {
```

Comment:
NPE if `listFiles` returns null, as it is permitted to do even if you believe it currently exists.